exporters/pelorus/config/__init__.py

```diff
@@ -0,0 +1,269 @@
+"""
+Unified, declarative configuration management and logging.
+
+Configuration needs to be consistent, and easy to get right.
+Configuration should be logged in order to make debugging easier.
+However, accidentally logging sensitive information (API credentials, etc.) must be hard.
+
+This module handles the above goals by using `attrs`.
+
+# A Simple Example
+
+```python
+from pelorus.config import config, load_and_log, var
+
+@config
+class MyConfiguration:
+    username: str = "GVR"
+    password: str
+    namespaces: list[str]
+```
+
+Calling `load_and_log(MyConfiguration)` will perform the following:
+
+1. Look for the environment variable `USERNAME`.
+2. Look for the environment variable `PASSWORD`.
+3. Look for the environment variable `NAMESPACES` and split at each comma, stripping all whitespace.
+4. Call `MyConfiguration(...)`:
+  If `USERNAME` is missing, it will default to `GVR`.
+  If `PASSWORD` is missing, a TypeError will be thrown.
+  If `NAMESPACES` is present but empty, an empty list will be given.
+
+It will then print `MyConfiguration` followed by each field's value.
+Sensitive values will be redacted.
+Each field will also list where the value came from.
+
+For example:
+
+1. `username=GVR default value; USERNAME was not set)`
+2. `password=REDACTED (from env var PASSWORD)`.
+  (Any field name that contains any word in `REDACT_WORDS` will not be printed by default.)
+3. `namespaces=[''] (from env var NAMESPACES)`
+
+# Supported Variable Types
+
+| type      | transformation from str                                 |
+|-----------|---------------------------------------------------------|
+| str       | set as-is                                               |
+| list[str] | split on `,` with whitespace stripped from each element |
+| set[str]  | same as above                                           |
+
+Optional versions of the above will always work, but the default must be set to `None`.
+
+Other types may be supported with customization and converters, shown in the next section.
+
+# Customization
+
+You may need to deviate from the default behavior, or pass an argument that won't come from the environment.
+You can customize each field with `var`:
+
+@config
+class AdvancedConfiguration:
+    anonymous_user: str = var(log=False)
+    should_pass_tests: bool = var(log=True)
+    whoami: str = var(env_lookups=["API_USER", "USER"])
+
+    api_client: Client = var(env_lookups=None)
+
+    optional_name: str = var(default="someone")
+    optional_list: list[str] = var(factory=list)
+
+    _private_field: Any = var(env_lookups=[])
+    public_but_ignore_me: Any = var(log=None)
+
+Notable differences:
+- anonymous_user will not be logged, when it otherwise would be.
+- should_pass_tests _will_ be logged.
+  We'd assume it's sensitive because it has "pass" in it, so we override that behavior.
+- whoami is set to the first value found of either `API_USER` or `USER`, instead of looking at `WHOAMI`.
```

Comment:
is or `USER` still true after https://github.com/konveyor/pelorus/pull/556 merges?